src/org/sosy_lab/java_smt/basicimpl/withAssumptionsWrapper/BasicProverWithAssumptionsWrapper.java

```diff
@@ -44,13 +60,16 @@ public void pop() {
 
   @Override
   public T addConstraint(BooleanFormula constraint) throws InterruptedException {
-    clearAssumptions();
+    // This might throw the "wrong" exception because it pops the assumptions
+    // (which does not return a InterruptedException for interrupts)
```

Comment:
Why does `clearAssumptionsWithInterruptedException()` throw a wrong exception? And why is there only this comment mentioning this instead of code that fixes it? And why only for this single call to `clearAssumptionsWithInterruptedException()`?